File: infra/src/main/java/com/beat/infra/external/sms/CoolSmsAdapter.java
package com.beat.infra.external.sms;

import com.beat.contracts.sms.SmsMessage;
import com.beat.contracts.sms.SmsPort;
import java.util.HashMap;
import lombok.RequiredArgsConstructor;
import net.nurigo.java_sdk.api.Message;
import net.nurigo.java_sdk.exceptions.CoolsmsException;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CoolSmsAdapter implements SmsPort {

	@Value("${spring.coolsms.api.key}")
	private String apiKey;

	@Value("${spring.coolsms.api.secret}")
	private String apiSecret;

	@Value("${spring.coolsms.api.number}")
	private String fromPhoneNumber;

	@Override
	public void sendSms(SmsMessage message) {
		String cleanedPhoneNumber = message.getTo().replace("-", "");
		Message coolsms = new Message(apiKey, apiSecret);

		HashMap<String, String> params = new HashMap<>();
		params.put("to", cleanedPhoneNumber);
		params.put("from", fromPhoneNumber);
		params.put("type", "SMS");
		params.put("text", message.getText());

		try {
			coolsms.send(params);
		} catch (CoolsmsException exception) {
			throw new RuntimeException("SMS 전송 실패: " + exception.getMessage(), exception);
		}
	}
}

File: infrastructure/src/main/kotlin/com/beat/infra/external/notification/sms/CoolSmsAdapter.kt
package com.beat.infra.external.notification.sms

import net.nurigo.java_sdk.api.Message
import net.nurigo.java_sdk.exceptions.CoolsmsException
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import java.util.HashMap

@Service
internal class CoolSmsAdapter {
    @field:Value("\${spring.coolsms.api.key}")
    private lateinit var apiKey: String

    @field:Value("\${spring.coolsms.api.secret}")
    private lateinit var apiSecret: String

    @field:Value("\${spring.coolsms.api.number}")
    private lateinit var fromPhoneNumber: String

    fun send(to: String, text: String) {
        val cleanedPhoneNumber = to.replace("-", "")
        val coolsms = Message(apiKey, apiSecret)
        val params =
            HashMap<String, String>().apply {
                put("to", cleanedPhoneNumber)
                put("from", fromPhoneNumber)
                put("type", "SMS")
                put("text", text)
            }

        try {
            coolsms.send(params)
        } catch (exception: CoolsmsException) {
            throw RuntimeException("SMS 전송 실패: ${exception.message}", exception)
        }
    }
}
